src/DotLiquid/StandardFilters.cs

```diff
@@ -166,12 +166,14 @@ public static string H(string input)
         /// <param name="length"></param>
         /// <param name="truncateString"></param>
         /// <returns></returns>
-        public static string Truncate(string input, int length = 50, string truncateString = "...")
+        public static string Truncate(string input, long length = 50, string truncateString = "...")
         {
             if (string.IsNullOrEmpty(input))
                 return input;
 
-            int l = length - truncateString.Length;
+            int lengthAsInt = length < int.MinValue ? int.MinValue : (length > int.MaxValue ? int.MaxValue : (int)length);

```

Comment:
A length input beyond int min or max appears to be an error. Hence the right behavior is to error out, rather than do a best effort to the nearest value in the range of int.

Also... what is the meaning of a negative length (int.MinValue) in this context? Any negative value may be an error condition already.

Reply:
Yes, fair enough. Will change to throw an ArgumentException.